Mathlib/LinearAlgebra/DirectSum/Basis.lean

```diff
@@ -0,0 +1,30 @@
+/-
+Copyright (c) 2021 Riccardo Brasca. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Riccardo Brasca
+-/
+import Mathlib.Algebra.DirectSum.Module
+import Mathlib.LinearAlgebra.Finsupp.VectorSpace
+
+/-!
+# Bases for direct sum of modules
```

Comment:
Should this file be renamed `Free.lean` instead? And the title rewritten accordingly?

Reply:
I named it `Basis.lean` analogously to `TensorProduct/Basis.lean`. I'm not sure if we want to keep seeing through the defeq and write `DFinsupp.basis` as the basis for `DirectSum`. If we want a separate `DirectSum.basis` definition, this would be the obvious file to put it. Or should we just name this `Free.lean` and rename it when we decide to refactor?

Reply:
Hmm, fair enough. I don't really care.

bors d+